src/lightning/pytorch/strategies/xla.py

```diff
@@ -304,6 +320,7 @@ def all_gather(self, tensor: Tensor, group: Optional[Any] = None, sync_grads: bo
 
     def teardown(self) -> None:
         super().teardown()
+        self._launched = False  # after the Trainer finishes, we aren't inside the spawned region
```

Comment:
This won't do anything. `self._launched` is always False outside the spawned region, and teardown runs inside.

Reply:
I know, but it can still be useful if one expects that `self.model` is wrapped in a call after teardown (such as checkpoint saving for xlafsdp)

Reply:
Yes, not saying it is wrong. Just wanted to make sure that we understand teardown() runs within the processes and so the main process never sees the `self._launched` True even without this change.